File: frontend/src/api/weeklyDigest.ts
import { api } from './client';

export interface DigestChildTask {
  completed: number;
  total: number;
}

export interface DigestChildAssignment {
  submitted: number;
  due: number;
}

export interface DigestQuizScore {
  quiz_count: number;
  average_percentage: number | null;
}

export interface DigestOverdueItem {
  id: number;
  title: string;
  due_date: string | null;
  item_type: string;
}

export interface ChildDigest {
  student_id: number;
  full_name: string;
  grade_level: number | null;
  tasks: DigestChildTask;
  assignments: DigestChildAssignment;
  study_guides_created: number;
  quiz_scores: DigestQuizScore;
  overdue_items: DigestOverdueItem[];
  highlight: string;
}

export interface WeeklyDigestResponse {
  week_start: string;
  week_end: string;
  greeting: string;
  children: ChildDigest[];
  overall_summary: string;
}

export interface WeeklyDigestSendResponse {
  success: boolean;
  message: string;
}

export const weeklyDigestApi = {
  preview: async () => {
    const response = await api.get('/api/digest/weekly/preview');
    return response.data as WeeklyDigestResponse;
  },
  send: async () => {
    const response = await api.post('/api/digest/weekly/send');
    return response.data as WeeklyDigestSendResponse;
  },
};

File: app/schemas/weekly_digest.py
"""Schemas for the Weekly Progress Pulse digest."""

from pydantic import BaseModel


class DigestChildTask(BaseModel):
    completed: int = 0
    total: int = 0


class DigestChildAssignment(BaseModel):
    submitted: int = 0
    due: int = 0


class DigestQuizScore(BaseModel):
    quiz_count: int = 0
    average_percentage: float | None = None


class DigestOverdueItem(BaseModel):
    id: int
    title: str
    due_date: str | None = None
    item_type: str  # "task" or "assignment"


class ChildDigest(BaseModel):
    student_id: int
    full_name: str
    grade_level: int | None = None
    tasks: DigestChildTask = DigestChildTask()
    assignments: DigestChildAssignment = DigestChildAssignment()
    study_guides_created: int = 0
    quiz_scores: DigestQuizScore = DigestQuizScore()
    overdue_items: list[DigestOverdueItem] = []
    highlight: str = ""  # one-line summary


class WeeklyDigestResponse(BaseModel):
    """JSON preview of the weekly digest for a parent."""
    week_start: str  # ISO date
    week_end: str  # ISO date
    greeting: str
    children: list[ChildDigest] = []
    overall_summary: str = ""


class WeeklyDigestSendResponse(BaseModel):
    success: bool
    message: str
